core/common/src/main/java/alluxio/resource/LockResource.java

```diff
@@ -33,8 +33,20 @@
    * @param lock the lock to acquire
    */
   public LockResource(Lock lock) {
+    this(lock, false);
+  }
+
+  /**
+   * Creates a new instance of {@link LockResource} using the given lock.
+   *
+   * @param lock the lock to acquire
+   * @param alreadyLocked whether the lock resource needs to be locked, or is already locked
+   */
+  public LockResource(Lock lock, boolean alreadyLocked) {
     mLock = lock;
-    mLock.lock();
+    if (!alreadyLocked) {
```

Comment:
I think `alreadyLocked` was misleading, so I've renamed it to `acquireLock`. It's not that we want to lock it only if it isn't already locked. What we really want is to lock it by default for convenience, but give the caller a way to specify that they haven't already acquire the lock so that we can use `LockResource` in conjunction with `tryLock`